### PR TITLE
Block mvp

### PR DESCRIPTION
**Pull request by Eein**
requires #106

### CODE REVIEW

File: memory/combat_manager.py [hunk 1]
```diff
@@ -55,6 +58,21 @@ def __init__(self: Self) -> None:
         self.spell_locks = []
 
 
+class NextCombatEnemy:
+    def __init__(self: Self) -> None:
```
[the remark this answers is not in the document]
Comment (Eein) [reply memory/combat_manager.py:62]: I can update this per your request - i didn't know what i was going to be storing here as i was working on it.

File: engine/combat/combat_controller.py [hunk 2]
```diff
@@ -59,6 +62,39 @@ def execute_combat(self: Self) -> bool:
...
+            next_combat_enemy = combat_manager.next_combat_enemy
+            if (
+                next_combat_enemy
+                and next_combat_enemy.state_type is NextCombatAction.Attacking
+                and next_combat_enemy.movement_done is True
+            ):
+                if self.block_timing is None:
+                    # get the block time from the enemy
+                    # This is where we calculate the time to block for the specific skill
+                    delta = timedelta(seconds=0.3)  # slug?
```
[the remark this answers is not in the document]
Comment (Eein) [reply engine/combat/combat_controller.py:80]: this specific delta is just the wait time from when the enemy stops moving and starts attacking - and this needs to be checked on every attack - will this be a problem?

it seems like coupling the combat controller to the sequencer delta may be a source of unintended behaviour..

[the remark this answers is not in the document]
Comment (orkaboy) [reply engine/combat/combat_controller.py:80]: The delta in the sequencer is just the time between two frames. So if you bring it in here, you can add it to an accumulated value until that reaches the target (0.3 seconds in this case). You'd still need to keep the accumulated time value in a variable here. Compare this to the mashing skip sequencer node, which does something similar.

File: engine/combat/combat_and_move.py [hunk 3]
```diff
@@ -32,6 +32,7 @@ def execute(self: Self, delta: float) -> bool:
                 if combat_manager.encounter_done is True:
                     self.state = EncounterState.POST_COMBAT
                 else:
+                    self.combat_controller.delta = delta
```
Comment (orkaboy) [inline engine/combat/combat_and_move.py:35]: another way is `self.combat_controller.execute_combat(delta)`

**Comment by shenef** [comment]
at the start of the wyrd fight:```
Traceback (most recent call last):
  File "E:\SoS-TAS\main.py", line 40, in <module>
    menu_manager.run()
  File "E:\SoS-TAS\GUI\menu.py", line 51, in run
    menu.run(top_level=True)
  File "E:\SoS-TAS\GUI\menu.py", line 36, in run
    done = self.active.run()
           ^^^^^^^^^^^^^^^^^
  File "E:\SoS-TAS\GUI\menu.py", line 35, in run
    return self.execute(top_level)
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "E:\SoS-TAS\route\TAS.py", line 73, in execute
    self.sequencer.run()
  File "E:\SoS-TAS\engine\seq\sequencer.py", line 95, in run
    self._update()
  File "E:\SoS-TAS\engine\seq\sequencer.py", line 67, in _update
    self.done = self.root.execute(delta=delta)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "E:\SoS-TAS\engine\seq\base.py", line 86, in execute
    ret = cur_child.execute(delta=delta)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "E:\SoS-TAS\engine\seq\base.py", line 86, in execute
    ret = cur_child.execute(delta=delta)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "E:\SoS-TAS\engine\seq\base.py", line 86, in execute
    ret = cur_child.execute(delta=delta)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "E:\SoS-TAS\engine\combat\combat_and_move.py", line 35, in execute
    self.combat_controller.execute_combat(delta)
TypeError: CombatController.execute_combat() takes 1 positional argument but 2 were given
```

**Pull request merged**